Cockpit 282 (#593)
cockpit-project/cockpit-project.github.io: _posts/2022-12-14-cockpit-282.md

---
title: Cockpit 282
author: garrett
date: '2022-12-14'
tags: cockpit
slug: cockpit-282
category: release

> summary: ''

Author: ```suggestion
summary: 'Accounts page redesign and right-to-left language support'
```

![screenshot of redesign and include groups]({{ site.baseurl }}/images/282-accounts-redesign-and-include-groups-2.png)

## Right-to-left language support

Right-to-left language support has been enabled for languages that require it. When selecting a RTL language from the list, the layout will shift to be right-aligned.

We have also started modifying Cockpit and PatternFly to work better with RTL languages. However, there are still some alignment and spacing issues. We will continue to work to fix the more obvious problems.

If you know Hebrew (Cockpit's most complete RTL translation) or other RTL languages, please help us improve the implementation and translations by [filing issues on GitHub](https://github.com/cockpit-project/cockpit/issues/new?labels=bug,i18n), sending pull requests, and/or [improving translations on weblate](https://translate.fedoraproject.org/projects/cockpit/main/).

## Try it out

Cockpit 282 is available now:

* [For your Linux system](https://cockpit-project.org/running.html)
* [Cockpit Client](https://flathub.org/apps/details/org.cockpit_project.CockpitClient)

* [Cockpit Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/282)

> * [Cockpit Fedora 37](https://bodhi.fedoraproject.org/updates/?releases=F37&packages=cockpit)
> * [Cockpit Fedora 36](https://bodhi.fedoraproject.org/updates/?releases=F36&packages=cockpit)

Author: ```suggestion
* [Cockpit Fedora 37](https://bodhi.fedoraproject.org/updates/FEDORA-2022-14210d1ef5)
* [Cockpit Fedora 36](https://bodhi.fedoraproject.org/updates/FEDORA-2022-9eabd9cf61)
```

## Accounts: Redesign and include groups

The Accounts page has been redesigned. Groups are listed at the top of the page and the list of users is enhanced with details and filtering. Additionally, a user's groups can be edited while viewing details for the user.

![screenshot of redesign and include groups]({{ site.baseurl }}/images/282-accounts-redesign-and-include-groups.png)

![screenshot of redesign and include groups]({{ site.baseurl }}/images/282-accounts-redesign-and-include-groups-2.png)

## Right-to-left language support

Right-to-left language support has been enabled for languages that require it. Our most complete RTL language is Hebrew. When choosing Hebrew in the languages list, the layout will shift and become right-aligned.

In addition to basic RTL support, we have started to modify Cockpit and PatternFly to work with RTL languages in a better way. This is not complete, but we wanted to enable this as soon as possible, so people could use it and report bugs.

We are aware that there are some basic issues across Cockpit. Most of Cockpit should be usable in RTL, but there are still some alignment and spacing problems.

The most affected pages at the moment are:

> - the login page (which does not currently adapt to RTL)

Reviewer: Not true. It works just fine (but you need to mount the static dir locally).

Author: Whoops! I forgot about the login screen being a special case. Thanks! Changed the intro based on this comment and our F2F discussion about it.

> Right-to-left language support has been enabled for languages that require it. Our most complete RTL language is Hebrew. When choosing Hebrew in the languages list, the layout will shift and become right-aligned.
> 
> In addition to basic RTL support, we have started to modify Cockpit and PatternFly to work with RTL languages in a better way. This is not complete, but we wanted to enable this as soon as possible, so people could use it and report bugs.
> 
> We are aware that there are some basic issues across Cockpit. Most of Cockpit should be usable in RTL, but there are still some alignment and spacing problems.
> 
> The most affected pages at the moment are:
> - the login page (which does not currently adapt to RTL)
> - optional pages such as `cockpit-podman`, `cockpit-machines`, and `cockpit-ostree`
> - all 3rd party app pages
> 
> If you know Hebrew, then please help us with the implementation. If you know other RTL languages, we're happy to work with you to help improve translations.
> 
> We will try to further improve RTL to the best of our ability, but need help from people who actually know Hebrew to make sure it is correct.

Author: ```suggestion
Right-to-left language support has been enabled for languages that require it. Our most complete RTL language is Hebrew. When choosing Hebrew in the languages list, the layout will shift and become right-aligned.

In addition to basic RTL support, we have started to modify Cockpit and PatternFly to work with RTL languages in a better way. This is not complete, but we wanted to enable this as soon as possible, so people could use it and report bugs.

We are aware that there are some basic issues across Cockpit. Most of Cockpit should be usable in RTL, but there are still some alignment and spacing problems.

We will try to further improve RTL to the best of our ability, but need help from people who actually know Hebrew to make sure it is correct. If you know Hebrew, then please help us with the implementation. If you know other RTL languages, we're happy to work with you to help improve translations.
```

Author: An even shorter version:

```suggestion
Right-to-left language support has been enabled for languages that require it. When selecting a RTL language from the list, the layout will shift to be right-aligned.

We have also started modifying Cockpit and PatternFly to work better with RTL languages. However, there are still some alignment and spacing issues. We will continue to work to fix the more obvious problems.

If you know Hebrew (Cockpit's most complete RTL translation) or other RTL languages, please help us improve the implementation and translations by [filing issues on GitHub](https://github.com/cockpit-project/cockpit/issues/new?labels=bug,i18n), sending pull requests, and/or [improving translations on weblate](https://translate.fedoraproject.org/projects/cockpit/main/).
```
